### PR TITLE
Add iconfinder

### PR DESCRIPTION
**Pull request by luis-pereira**
It helps testing and evaluating https://github.com/lxde/libqtxdg/pull/116.

### CODE REVIEW

File: util/qtxdg-iconfinder.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    parser.addVersionOption();
+    parser.addHelpOption();
+    parser.process(app);
+
+    if (parser.positionalArguments().isEmpty())
+        parser.showHelp(EXIT_FAILURE);
+
+    const QStringList icons = parser.positionalArguments();
+    for (const QString& iconName : icons) {
+        const auto icon = XdgIconLoader::instance()->loadIcon(iconName).iconName;
```
Comment (palinek) [inline util/qtxdg-iconfinder.cpp]: I would call the `XdgIconLoader::loadIcon()` only once...

File: util/qtxdg-iconfinder.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        const auto entries = XdgIconLoader::instance()->loadIcon(iconName).entries;
+
+        std::cout << qPrintable(iconName) <<
+            qPrintable(QString::fromLatin1(":")) << qPrintable(icon) << "\n";
+
+        if (!icon.isEmpty()) {
+            if (!entries.empty()) {
```
Comment (palinek) [inline util/qtxdg-iconfinder.cpp]: Is the `empty()` check necessary?

File: util/qtxdg-iconfinder.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                const auto end = entries.constEnd();
+                for (auto i = entries.constBegin(); i != end; ++i) {
```
Comment (palinek) [inline util/qtxdg-iconfinder.cpp]: Just curious, why are you using the for range loop on one place and manual iteration on other?

File: util/qtxdg-iconfinder.cpp (hunk 1 above)
Comment (luis-pereira) [inline util/qtxdg-iconfinder.cpp]: Me too. I added the entries stuff later, and .... bed prevent me from compacting it.

File: util/qtxdg-iconfinder.cpp (hunk 2 above)
Comment (luis-pereira) [inline util/qtxdg-iconfinder.cpp]: Reasoning in the former comment.

File: util/qtxdg-iconfinder.cpp (hunk 3 above)
Comment (luis-pereira) [inline util/qtxdg-iconfinder.cpp]: I'm trying to adopt the range for and other C++11 stuff... but years of using the manual iterators "sometimes" kick in, without I even notice it.

**Comment by luis-pereira** [comment]
@palinek Updated.

File: util/qtxdg-iconfinder.cpp (hunk 2 above)
Comment (palinek) [inline util/qtxdg-iconfinder.cpp]: I meant the `if (!entries.empty())` ...because if the `entries` are empty, the for loop will do nothing.

**Comment by luis-pereira** [comment]
@palinek Thanks. Jet-lag is killing me.

**Pull request merged**